lxd/images.go

```diff
@@ -353,18 +381,15 @@ func imageDelete(d *Daemon, r *http.Request) Response {
 	return EmptySyncResponse
 }
 
-func imageGet(d *Daemon, r *http.Request) Response {
-	fingerprint := mux.Vars(r)["fingerprint"]
-
-	public := !d.isTrustedClient(r)
+func doImageGet(d *Daemon, fingerprint string, public bool) (shared.ImageInfo, Response) {
 	imgInfo, err := dbImageGet(d, fingerprint, public)
 	if err != nil {
-		return SmartError(err)
+		return shared.ImageInfo{}, SmartError(err)
```

Comment:
On Mon, Apr 06, 2015 at 03:37:45PM -0700, St�phane Graber wrote:
> >  	imgInfo, err := dbImageGet(d, fingerprint, public)
> >  	if err != nil {
> > -		return SmartError(err)
> > +		return shared.ImageInfo{}, SmartError(err)
> 
> Oh, I certainly tried that, but Go wouldn't let me :)
> 
> ../../go/src/github.com/lxc/lxd/lxd/images.go:392: cannot use nil as type shared.ImageInfo in return argument

Oh, right, has to be a pointer D:

> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/469/files#r27837475